### PR TITLE
SQL ORM: Data Ingestion into Database

### PR DESCRIPTION
**Pull request by GbotemiB**
This PR includes the implementation of data ingestion into a Postgres database using Object Relational Mapping (ORM) in Python. 

The data set used for the implementation can be found [here](https://github.com/Data-Epic/Shadow-Ops/raw/main/project-4/data/cleaned_artwork_data.parquet)

### CODE REVIEW

File: project-6/project_6/main.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+class Artwork(Base):
+    """
+    A class to create an Artwork table using ORM classes
+    """
+
+    __tablename__ = "artwork"
+    __table_args__ = {"extend_existing": True}
+
+    ConstituentID = Column(Integer, primary_key=True, index=True)
+    DisplayName = Column(String, nullable=False)
```
Comment (elseagle) [inline project-6/project_6/main.py]: Will these columns translate to snake_case? If not, snake_case and lower case, please

File: project-6/project_6/main.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+warnings.filterwarnings("ignore")
+
+load_dotenv()
+logging.basicConfig(level=logging.INFO)
+
+Base = declarative_base()
+
+
+# create orm sql classes
+class Artwork(Base):
```
Comment (elseagle) [inline project-6/project_6/main.py]: nit: `ArtWork` ?

File: project-6/project_6/main.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            logging.info("data ingestion into database started")
+            # validate the row before inserting into db
+            if type_check(row):
+                line = Artwork()
+                for key, value in row.items():
+                    setattr(line, key, value)
+                session.add(line)
+                session.commit()
+            else:
+                logging.error(f"type check failed for {idx[0]}:{row}")
```
Comment (elseagle) [inline project-6/project_6/main.py]: upper case for first letter - apply to all logs

File: project-6/project_6/main.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        except IntegrityError as e:
+            logging.error(e)
+            session.rollback()
+
+    logging.info("data ingestion completed")
+
+
+if __name__ == "__main__":
+    # load environment variables
+    username = os.environ.get("username")
```
Comment (elseagle) [inline project-6/project_6/main.py]: UPPER_CASE For env vars please

File: project-6/project_6/query_db.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+# load environment variables
+load_dotenv()
+
+username = os.environ.get("username")
+password = os.environ.get("password")
+host = os.environ.get("host")
+
+# prepare connection to db
+engine = create_engine(f"postgresql+psycopg2://{username}:{password}@{host}")
```
Comment (elseagle) [inline project-6/project_6/query_db.py]: move DB_URI to env

File: project-6/project_6/query_db.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            func.count(Artwork.Department).desc())
+    logging.info(f"Departments : {query.all()}")
+
+# top 5 medium used in the database
+with Session(bind=engine) as session:
+    query = session.query(Artwork.Medium, func.count(Artwork.Medium)).group_by(
+        Artwork.Medium).order_by(func.count(Artwork.Medium).desc())
+    logging.info(f"Top 5 Medium used : {query.all()[:5]}")
+
+# gender distribution in database
```
Comment (elseagle) [inline project-6/project_6/query_db.py:46]: Love the queries!

File: project-6/tests/test_main.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+from unittest.mock import patch, Mock
+
+
+def test_load_data_from_file():
+    # assert if the result is a pd.dataframe
+    pass
+
+
+def test_type_check():
+    test_dict = {
```
Comment (elseagle) [inline project-6/tests/test_main.py:14]: use `pytest.fixture` and `conftest.py` for data loading.

**Review by elseagle — CHANGES_REQUESTED** [review]
Solid work! Some changes requested

File: project-6/project_6/query_db.py (hunk 6 above)
Comment (GbotemiB) [reply project-6/project_6/query_db.py:46]: thank you

File: project-6/project_6/main.py (hunk 1 above)
Comment (GbotemiB) [reply project-6/project_6/main.py]: I have changed the values to snake_case

File: project-6/project_6/main.py (hunk 2 above)
Comment (GbotemiB) [reply project-6/project_6/main.py]: Thanks for the observation

File: project-6/project_6/main.py (hunk 3 above)
Comment (GbotemiB) [reply project-6/project_6/main.py]: All Done

File: project-6/project_6/main.py (hunk 4 above)
Comment (GbotemiB) [reply project-6/project_6/main.py]: All done

**Review by elseagle — APPROVED** [review]

**Pull request merged**